src/Promise/PromiseInterface.php

```diff
@@ -0,0 +1,29 @@
+<?php
+namespace GraphQL\Promise;
+
+interface PromiseInterface
+{
+    /**
+     * Waits until the promise completes if possible.
+     *
+     * @return mixed
+     * @throws \LogicException if the promise has no wait function.
+     */
+    public function wait();
+
```

Comment:
Hey I like your library - it's interface is similar to what I imagined. So we are likely thinking in the same direction.

But I'd like to avoid external dependencies for GraphQL lib. Actually I considered adding very simple and limited implementation for our sync case as most of the users do not need full-featured promises in sync environment.

So I'd really like to have some abstraction for Promises as a part of this library.  We can add your lib to composer `suggests` entry for those who need full-blown promise support.